### PR TITLE
Extensions for core and storage k8s functionality.

### PR DESCRIPTION
**Pull request by igomez06**
Core and storage k8s helpers
**Framework:**
Implemented new client functionality to be able to get a downstream k8s client.
Added helpers for volumes (persistent volumes and persistent volume claims), storage classes, secrets, and config maps

### CODE REVIEW

File: tests/framework/extensions/secrets/basicAuth.go [hunk 1]
```diff
@@ -0,0 +1,24 @@
+package secrets
+
+import (
+	coreV1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func NewBasicAuthSecret(name, namespace, username, password string) *coreV1.Secret {
+	return &coreV1.Secret{
+		TypeMeta: metav1.TypeMeta{
```
Comment (paynejacob) [inline tests/framework/extensions/secrets/basicAuth.go:10]: This is redundant, the k8s client will derive this from the struct type.

File: tests/framework/extensions/configmaps/configmaps.go [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+	Resource: "configmaps",
+}
+
+// CreateConfigMap is a helper function that uses the dynamic client to create a config map on a namespace for a specific cluster.
+// It registers a delete fuction.
+func CreateConfigMap(client *rancher.Client, clusterName, configMapName, description, namespace string, data, labels, annotations map[string]string) (*coreV1.ConfigMap, error) {
+	// ConfigMap object for a namespace in a cluster
+	annotations["field.cattle.io/description"] = description
+	configMap := &coreV1.ConfigMap{
+		TypeMeta: metav1.TypeMeta{
```
Comment (paynejacob) [inline tests/framework/extensions/configmaps/configmaps.go:27]: This is redundant, the k8s client will derive this from the struct type.

**Review by paynejacob — COMMENTED** [review]
The helpers for the corev1 resources seem abstract is there a specific use case these are being created for?

File: tests/framework/extensions/configmaps/configmaps.go [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+	}
+
+	configMapResource := dynamicClient.Resource(ConfigMapGroupVersionResource).Namespace(namespace)
+
+	unstructuredResp, err := configMapResource.Create(context.TODO(), unstructured.MustToUnstructured(configMap), metav1.CreateOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	client.Session.RegisterCleanupFunc(func() error {
```
Comment (paynejacob) [inline tests/framework/extensions/configmaps/configmaps.go:52]: Doesn't the dynamic client already do this?

File: tests/framework/extensions/secrets/secrets.go [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+	}
+
+	secretResource := dynamicClient.Resource(SecretGroupVersionResource).Namespace(namespace)
+
+	unstructuredResp, err := secretResource.Create(context.TODO(), unstructured.MustToUnstructured(secret), metav1.CreateOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	client.Session.RegisterCleanupFunc(func() error {
```
Comment (paynejacob) [inline tests/framework/extensions/secrets/secrets.go:36]: Doesn't the dynamic client already do this?

File: tests/framework/extensions/storageclasses/awsebs/create.go [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+	}
+
+	storageClassVolumesResource := dynamicClient.Resource(storageclasses.StorageClassGroupVersionResource).Namespace("")
+
+	unstructuredResp, err := storageClassVolumesResource.Create(context.TODO(), unstructuredStorageClass, metav1.CreateOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	client.Session.RegisterCleanupFunc(func() error {
```
Comment (paynejacob) [inline tests/framework/extensions/storageclasses/awsebs/create.go:50]: Doesn't the dynamic client already do this?

File: tests/framework/extensions/configmaps/configmaps.go (hunk 3 above)
Comment (igomez06) [reply tests/framework/extensions/configmaps/configmaps.go:52]: Yes this is true, this not needed

File: tests/framework/extensions/secrets/secrets.go (hunk 4 above)
Comment (igomez06) [reply tests/framework/extensions/secrets/secrets.go:36]: Yes this is true, this not needed

File: tests/framework/extensions/storageclasses/awsebs/create.go (hunk 5 above)
Comment (igomez06) [reply tests/framework/extensions/storageclasses/awsebs/create.go:50]: Yes this is true, this not needed

**Comment by igomez06** [comment]
> The helpers for the corev1 resources seem abstract is there a specific use case these are being created for?

Well just to make it easier on the developer when creating these resources, so then they can just make a single create call for these. Instead of having to convert the structs to unstructures every time when wanted create a config map for example. I mean I think having the constructors for some of these brings some value. But I'm open to reimplement.

File: tests/framework/extensions/secrets/basicAuth.go (hunk 1 above)
Comment (igomez06) [reply tests/framework/extensions/secrets/basicAuth.go:10]: I don't understand what you mean. There are different types of secrets, my thought process was make that easier to create for the developer.

File: tests/framework/extensions/configmaps/configmaps.go (hunk 2 above)
Comment (igomez06) [reply tests/framework/extensions/configmaps/configmaps.go:27]: Do you mean doing the TypeMeta is redundant?

**Review by paynejacob — APPROVED** [review]

**Pull request merged**